packages/flutter/lib/src/widgets/app.dart

```diff
@@ -24,18 +25,31 @@ import 'widget_inspector.dart';
 
 export 'dart:ui' show Locale;
 
+/// The signature of [WidgetsApp.localeListResolutionCallback].
+///
+/// A [LocaleListResolutionCallback] is responsible for computing the locale of the app's
+/// [Localizations] object when the app starts and when user changes the list of
+/// locales for the device.
+///
+/// The [locales] list is the device's preferred locales when the app started, or the
+/// device's preferred locales the user selected after the app was started. This list
+/// is in order of preference The [supportedLocales] parameter is just the value of
+/// [WidgetsApp.supportedLocales].
+typedef LocaleListResolutionCallback = Locale Function(List<Locale> locales, Iterable<Locale> supportedLocales);
+
 /// The signature of [WidgetsApp.localeResolutionCallback].
 ///
-/// A `LocaleResolutionCallback` is responsible for computing the locale of the app's
+/// A [LocaleResolutionCallback] is responsible for computing the locale of the app's
```

Comment:
from the docs it's unclear to me how this one is different from LocaleListResolutionCallback? Both should probably have a "See also" section that link to each other and explain the difference.